### ISSUE TITLE
Customizers aren't being used?

### ISSUE THREAD

**Issue by hildo** [infinispan/infinispan-spring-boot#43]
Hi.  I'm using 2.0.0 of the starter, and I've been able to use the starter to set up an EmbeddedCacheManager.  When I attempt to run my tests (which repeatedly start and stop my processes), I get a bunch of exceptions like this, which prevent my spring boot process to start more than once...

```
testSaveUnknownArea(minestar.core.site.rest.internal.plan.ProductionTargetInternalRestControllerSlowTest)  Time elapsed: 0.016 sec  <<< ERROR!
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'userAuthenticationConverterServiceImpl': Unsatisfied dependency expressed through method 'setEmbeddedCacheManager' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'defaultCacheManager' defined in class path resource [org/infinispan/spring/starter/embedded/InfinispanEmbeddedAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.infinispan.manager.DefaultCacheManager]: Factory method 'defaultCacheManager' threw exception; nested exception is org.infinispan.jmx.JmxDomainConflictException: ISPN000034: There's already a JMX MBean instance type=CacheManager,name="DefaultCacheManager" already registered under 'infinispan' JMX domain. If you want to allow multiple instances configured with same JMX domain enable 'allowDuplicateDomains' attribute in 'globalJmxStatistics' config element
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'defaultCacheManager' defined in class path resource [org/infinispan/spring/starter/embedded/InfinispanEmbeddedAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.infinispan.manager.DefaultCacheManager]: Factory method 'defaultCacheManager' threw exception; nested exception is org.infinispan.jmx.JmxDomainConflictException: ISPN000034: There's already a JMX MBean instance type=CacheManager,name="DefaultCacheManager" already registered under 'infinispan' JMX domain. If you want to allow multiple instances configured with same JMX domain enable 'allowDuplicateDomains' attribute in 'globalJmxStatistics' config element
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.infinispan.manager.DefaultCacheManager]: Factory method 'defaultCacheManager' threw exception; nested exception is org.infinispan.jmx.JmxDomainConflictException: ISPN000034: There's already a JMX MBean instance type=CacheManager,name="DefaultCacheManager" already registered under 'infinispan' JMX domain. If you want to allow multiple instances configured with same JMX domain enable 'allowDuplicateDomains' attribute in 'globalJmxStatistics' config element
Caused by: org.infinispan.jmx.JmxDomainConflictException: ISPN000034: There's already a JMX MBean instance type=CacheManager,name="DefaultCacheManager" already registered under 'infinispan' JMX domain. If you want to allow multiple instances configured with same JMX domain enable 'allowDuplicateDomains' attribute in 'globalJmxStatistics' config element
```

It looks like JMX errors, which I don't really need.  So I've attempted to use a GlobalConfigurationCustomizer to disable JMX as requested.  I've created a Spring Bean like this:

![image](https://user-images.githubusercontent.com/1204224/37127899-7effc2ee-22c3-11e8-9bca-0abcae18cbfa.png)

My class is using @Configuration annotation at the top.  However, nothing has changed.  When I debug through InfinispanEmbeddedAutoConfiguration, my global configurer is not in the List globalCOnfigurationCustomizers.  It's always the Collections.emptyList() that is initialized.

How should I get the AutoConfiguration to pick up my customizer?  Am I doing something wrong?

**Comment by hildo** [comment]
I have been able to work around this issue by defining the global configuration within an XML file and use the XML file instead of the Spring Beans.  But I would prefer to use Spring Beans (as is the preference in Spring Boot applications) to achieve this instead.

**Comment by hildo** [comment]
The problem here was not the AutoConfiguration.  The root cause of this issue was how my test application was configured to scan for Configuration classes.  In my "proper" application, the customizers were picked up fine because my configuration class was loaded.  In my Spring Test / Junit tests where I was having the problem, however, it didn't include my configuration class.

Once I update my Test context to include my Configuration class, my configurer can be loaded up and the tests execute fine again.

**Comment by galderz** [comment]
This `org.infinispan.jmx.JmxDomainConflictException` issue has been fixed in Infinispan 9.2.0.Final. I also did a [blog post](http://blog.infinispan.org/2018/01/easier-infinispan-clusters-on-same-jvm.html) about it a couple of months back.
